tests/integration_tests/security/test_custom_seccomp.py

```diff
@@ -2,179 +2,76 @@
 """Tests that the --seccomp-filter parameter works as expected."""
 
-import os
 import platform
-import tempfile
 import time
+from pathlib import Path
 
-import pytest
 import requests
 
 from framework import utils
-from host_tools.cargo_build import run_seccompiler_bin
 
 
-def _custom_filter_setup(test_microvm, json_filter):
-    json_temp = tempfile.NamedTemporaryFile(delete=False)
-    json_temp.write(json_filter)
-    json_temp.flush()
+def install_filter(microvm, bpf_path):
+    """Install seccomp filter in microvm."""
+    microvm.create_jailed_resource(bpf_path)
+    microvm.jailer.extra_args.update({"seccomp-filter": bpf_path.name})
 
-    bpf_path = os.path.join(test_microvm.path, "bpf.out")
 
-    run_seccompiler_bin(bpf_path=bpf_path, json_path=json_temp.name)
+def test_allow_all(uvm_plain, seccompiler):
+    """Test --seccomp-filter, allowing all syscalls."""
+    seccomp_filter = {
+        thread: {"default_action": "allow", "filter_action": "trap", "filter": []}
+        for thread in ["vmm", "api", "vcpu"]
+    }
 
-    os.unlink(json_temp.name)
-    test_microvm.create_jailed_resource(bpf_path)
-    test_microvm.jailer.extra_args.update({"seccomp-filter": "bpf.out"})
-
-
-def _config_file_setup(test_microvm, vm_config_file):
-    test_microvm.create_jailed_resource(test_microvm.kernel_file)
-    test_microvm.create_jailed_resource(test_microvm.rootfs_file)
-
-    vm_config_path = os.path.join(test_microvm.path, os.path.basename(vm_config_file))
-    with open(vm_config_file, encoding="utf-8") as f1:
-        with open(vm_config_path, "w", encoding="utf-8") as f2:
-            for line in f1:
-                f2.write(line)
-    test_microvm.create_jailed_resource(vm_config_path)
-    test_microvm.jailer.extra_args = {"config-file": os.path.basename(vm_config_file)}
-
-    test_microvm.jailer.extra_args.update({"no-api": None})
-
-
-def test_allow_all(uvm_plain):
-    """
-    Test --seccomp-filter, allowing all syscalls.
-    """
+    bpf_path = seccompiler.compile(seccomp_filter)
     test_microvm = uvm_plain
-
-    _custom_filter_setup(
-        test_microvm,
-        """{
-        "Vmm": {
-            "default_action": "allow",
-            "filter_action": "trap",
-            "filter": []
-        },
-        "Api": {
-            "default_action": "allow",
-            "filter_action": "trap",
-            "filter": []
-        },
-        "Vcpu": {
-            "default_action": "allow",
-            "filter_action": "trap",
-            "filter": []
-        }
-    }""".encode(
-            "utf-8"
-        ),
-    )
-
+    install_filter(test_microvm, bpf_path)
     test_microvm.spawn()
-
     test_microvm.basic_config()
-
     test_microvm.start()
-
     utils.assert_seccomp_level(test_microvm.firecracker_pid, "2")
 
 
-def test_working_filter(uvm_plain):
-    """
-    Test --seccomp-filter, rejecting some dangerous syscalls.
-    """
-    test_microvm = uvm_plain
+def test_working_filter(uvm_plain, seccompiler):
```

Comment:
what exactly is the purpose of this test, actually? We use a custom filter that specifically allows everything but clone and exec, and then verify firecracker starts. But this filter is significantly more lax than our default filters, so what would this test actually catch? If we start using clone and exec, all other tests would fail already, since these calls are already disallowed by the default filters

Reply:
Yeah not sure this test catches anything we wouldn't catch already.